include/rcpputils/unique_lock.hpp

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCPPUTILS__UNIQUE_LOCK_HPP_
+#define RCPPUTILS__UNIQUE_LOCK_HPP_
+
+#include <mutex>
+
+#include "rcpputils/thread_safety_annotations.hpp"
+
+namespace rcpputils
+{
+
+/**
+ * @brief Trivial std::unique_lock wrapper providing constructor that allows Clang's
+ * Thread Safety Analysis.
+ * The libc++ std::unique_lock does not have these annotations.
+ */
+template<typename MutexT>
+class RCPPUTILS_TSA_SCOPED_CAPABILITY unique_lock : public std::unique_lock<MutexT>
+{
+public:
+  explicit unique_lock(std::mutex & mu) RCPPUTILS_TSA_ACQUIRE(mu)
```

Comment:
I'm not 100% sure of this, but shouldn't this be:

```suggestion
  explicit unique_lock(MutexT & mu) RCPPUTILS_TSA_ACQUIRE(mu)
```

Reply:
Yep, I need to add a test with two kinds of mutex before we merge this!